planex/spec.py

```diff
@@ -319,21 +319,30 @@ def load(specpath, link=None, check_package_name=True, defines=None):
         elif link.schema_version == 3:
             for name, value in link.sources.items():
                 idx = _parse_name(name)
-                if value["URL"].startswith("ssh://"):
+                if not value:
+                    spec.rm_source(idx)
+                elif value["URL"].startswith("ssh://"):
                     spec.add_gitarchive(idx, value["URL"],
                                         link.path, value.get("prefix"),
                                         value.get("commitish"))
                 else:
                     spec.add_source(idx, value["URL"], link.path)
             for name, value in link.patch_sources.items():
                 idx = _parse_name(name)
-                spec.add_archive(idx, value["URL"], link.path,
-                                 value["patches"])
+                if not value:
+                    spec.rm_patch(idx)
```

Comment:
Why does `rm_patch` match with `add_archive` here?   If that's what is intended could you please add a comment explaining why these two go together.